### DISCUSSION TITLE
How to draw T.MyProperty of LineSeries<T>?

### DISCUSSION THREAD

**Discussion by GF-Huang** [Live-Charts/LiveCharts2#256] [Q&A] [answered]
How to draw `MyClass.MyProperty` instead of `MyClass` itself?

```cs
public class MyClass
{
    public double MyProperty { get; set; }
}

public class ViewModel
{
    public ObservableCollection<MyClass> MyClassCollection { get; }

    public IEnumerable<ISeries> Series { get; }

    public ViewModel(ObservableCollection<MyClass> myClasses)
    {
        MyClassCollection = myClasses;
        
        Series = new ObservableCollection<ISeries>
	    {
	        new LineSeries<MyClass>
	        {
	            Values = MyClassCollection 
	        }
	    };
    }
}
```

**Comment by waffle-lord** [comment, marked as answer]
I believe you can use a mapper to do this.

https://github.com/beto-rodriguez/LiveCharts2/blob/master/docs/overview/1.5.mappers.md

**Reply by GF-Huang** [reply]
Why I got this exception?

![image](https://user-images.githubusercontent.com/4510984/138569169-73711b8c-5514-4d51-acbb-7b4ceb0d2890.png)

![image](https://user-images.githubusercontent.com/4510984/138569189-785a9997-d8ae-4463-81da-b7508956c04f.png)

**Reply by beto-rodriguez** [reply]
It seems like a bug to me, for know a workaround is to try:

```
config
  .AddDefaultMappers()
  .AddSkiaSharp()
  .AddLightTheme()
  .HasMap<TimeShare>(  ... );
```

**Reply by GF-Huang** [reply]
It works, thank you.

And how to set a `TimeSpan` to `SecondaryValue`? In other words, use `TimeSpan` as X-Axis.

**Reply by beto-rodriguez** [reply]
Use the ticks property, the following link is the default implementation:

[link](https://github.com/beto-rodriguez/LiveCharts2/blob/18adf1b555672289bcc89afb506236b43d44600e/src/LiveChartsCore/Kernel/LiveChartsSettings.cs#L416)

And here is an example consuming that type:

https://github.com/beto-rodriguez/LiveCharts2/blob/master/samples/ViewModelsSamples/Axes/TimeSpanScaled/ViewModel.cs

**Reply by GF-Huang** [reply]
It seems incorrect.

![image](https://user-images.githubusercontent.com/4510984/138569999-639cd879-4d20-4ee8-9c5a-3e5def915ddb.png)

**Reply by beto-rodriguez** [reply]
See the linked samples to format the axis.

**Reply by GF-Huang** [reply]
> It seems like a bug to me, for know a workaround is to try:
> 
> ```
> config
>   .AddDefaultMappers()
>   .AddSkiaSharp()
>   .AddLightTheme()
>   .HasMap<TimeShare>(  ... );
> ```

Seems not fixed by `beta.101`.
